### PR TITLE
MAINT: Compat with MNE 1.8 birthday date

### PR DESCRIPTION
**Pull request by larsoner**
1. Fix compat with MNE 1.8+
2. Require pytest>=8 (tests will fail on < 8)
3. Add `--fix` to autofix stuff like import order on attempted commit (saves time, follows MNE)
4. Add another warning skip that I got on matplotlib locally

### CODE REVIEW

**Comment by larsoner** [comment]
Needs https://github.com/mne-tools/mne-python/pull/12732 then should be green 🤞

**Review by sappelhoff — APPROVED** [review]
Thanks Eric!

**Comment by larsoner** [comment]
Failures are due to https://github.com/sphinx-doc/sphinx/issues/12589. I changed a bunch of `autolink` refs to explicit refs but couldn't find the last two. So for now I added a `!=7.4` which can be removed (or made more specific but it will be annoyingly long) once 7.4.7 comes out.

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,7 +48,13 @@ jobs:
...
             no_output_timeout: 30m
             command: |
-              make build-doc
+              make build-doc 2>&1 | tee sphinx_log.txt
+
+        - run:
+            name: Check sphinx log for warnings (which are treated as errors)
+            when: always
+            command: |
+              ! grep "^.*WARNING: .*$" sphinx_log.txt
```
Comment (sappelhoff) [inline .circleci/config.yml:57]: this is great, thanks!

**Review by sappelhoff — COMMENTED** [review]
thanks for the fixes and for reporting this at the sphinx issue!

**Pull request merged**